src/hyperGard.js

```diff
@@ -505,10 +505,23 @@ var
             });
           }
 
+          /**
+           * If flag to returnRawData is false & the object is a valid
+           * use Resource constructor
+           * @param {} data
+           */
+          function formatData(data) {
+            if (!o.returnRawData && isObject(data)) {
+              return new Resource(url, data);
+            }
+
+            return data;
+          }
+
           return response.json().then(function(data) {
             return {
               action: name,
-              data: isObject(data) ? new Resource(url, data) : response.text(),
```

Comment:
I removed the call to `response.text`, since this stream can not actually be read again at this point. `json` has already read the stream.